### ISSUE TITLE
Please create an AppData file for AssualtCube

### ISSUE THREAD

**Issue by hughsie** [assaultcube/AC#86]
Please consider writing and installing an AppData file with the application description and some screenshots, else AssualtCube looks really bad in the GNOME and KDE Software Centers. We'd love to showcase more applications, but without the extra data file we can't.

See http://people.freedesktop.org/~hughsient/appdata/ for details; thanks!

Richard

**Comment by zeSKB** [comment]
Hi there,

Thanks for bringing this to our attention.

This looks, however, like the job for distribution package maintainers. Is there any merit to this, when a piece of software comes (as a tarball) from somewhere, other than distribution repositories? There is some for desktop files, but for AppData, sadly, I don't see any.

Sarunas

**Comment by hughsie** [comment]
Just like it doesn't make sense for distros to provide .desktop files for an application, I don't think it makes sense for Suse to write an AppData file, and then Fedora to write just the same content. You as the upstream maintainer are in a position where you can take good screenshots of the parts of the application you're proudest of, and you're free to write a good description that is going to encourage people to install your app. You're probably in a position where you can localize the AppData file into multiple languages, which is something the distros just can't do.

Lots of desktops and distros are using AppStream now, and by not including an AppData file your application is sorted much lower down in the search results, and for GNOME 3.18, not shown at all.

**Comment by zeSKB** [comment]
You bring up a good point. I will try adding a few example files for distro maintainers.

**Comment by GreenLunar** [comment]
Make it translatable, too (#126).

**Comment by GreenLunar** [comment]
I have made an initial AppData file. I used appdata.xml from /freedoom/tree/master/dist of the Freedom project as a template.

```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Copyright 2015-2016 Contributors to the AssaultCube project -->

<component type="desktop">
  <id>assaultcube.desktop</id>
  <metadata_license>LICENSE</metadata_license>
  <project_license>Zlib/libpng</project_license>
  <developer_name>Adrian 'driAn' Henke, Andrew 'Arghvark' Davis, Lee 'Eihrul' Salzman, stef</developer_name>
  <url type="homepage">http://assault.cubers.net/</url>
  <url type="bugtracker">https://github.com/assaultcube/AC/issues</url>
  <description>
    <p>
      AssaultCube is a FREE, multiplayer, first-person shooter game,
      based on the CUBE engine.
    </p>
    <p>
      Taking place in realistic environments, with fast, arcade
      gameplay, it's addictive and fun! 
    </p>
    <p>
      With efficient bandwidth usage, it's low-latency and can even
      run over a 56Kbps connection.
      It's tiny too, weighing in at a lightweight 40 MB package.
      On the correct settings, it can even run on old hardware
      (Pentium III and above).
    </p>
  </description>
  <screenshots>
    <screenshot type="default">
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_complex_DM_gren_large00.jpg</image>
      <caption>ac_complex: A player accidentally blowing himself up with his own grenade.</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_complex_DM_pist_large00.jpg</image>
      <caption>ac_complex: Fragging another player with the akimbo power up (dual pistols).</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_complex_DM_SG_large00.jpg</image>
      <caption>ac_complex: A shotgun at short range.</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_complex_DM_SR_large00.jpg</image>
      <caption>ac_complex: Being fired at by the enemy, while scoping-in on the enemy and firing upon them.</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_mines_CTF_SMG_large00.jpg</image>
      <caption>ac_mines: A sub-machine gun on ac_mines.</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_mines_CTF_SMG_large01.jpg</image>
      <caption>ac_mines: You've just scored, and an enemy is already trying to capture your flag!</caption>
    </screenshot>
    <screenshot>
      <image>http://assault.cubers.net/screenshots/screenshots-large/ac_shine_CTF_unknown_large00.jpg</image>
      <caption>ac_shine: This is why you don't camp! (Special thanks to jamz for this image)</caption>
    </screenshot>
  </screenshots>
</component>
```

**Comment by GreenLunar** [comment]
Dear @hughsie

In concern to language, I disagree with the following comment.

> `<description/>`
> * Don't mention what language an application is written in, users don't care

I think, it would be proper to have a tag that would list available languages of application or, at least, have a Boolean tag that would indicate whether an application is multilingual or not.

**Comment by hughsie** [comment]
Language in that case was supposed to mean Java, C++, etc rather than the locales supported. When we convert the package to AppStream data we import the .po files and work out what locales are supported. If the translations are more than 25% complete compared to the largest language then we mark it as available in the language. If the users locale matches the supported locales when searching, the application is moved up the search results by a large amount. If you're not using .po files, we probably need a way to work out what languages are supported.

**Comment by GreenLunar** [comment]
This is good.

> If you're not using .po files, we probably need a way to work out what languages are supported.

Are .ts files supported?

**Comment by hughsie** [comment]
Presumably you're installing the .qm files, rather than the .ts file, right? Either way, no :) I need to look at a way to read the .qm file format, or to convert it to something like .po so the existing analysis tools work.

**Comment by GreenLunar** [comment]
> Presumably you're installing the .qm files, rather than the .ts file, right?

Yes; and, consequently, .mo files, rather than the .po file.

To make this clear to me, since you referred to .qm file format, would it be proper to assume that it is .mo file format (compiled .po) that is actually being read, using `msgunfmt`, I guess?

**Comment by hughsie** [comment]
Yes, we actually read the .mo file (unpacking the simple dictionary manually) as we only have the binary "built" rpms available to us.

**Comment by GreenLunar** [comment]
(this comment will be deleted)

May I continue discussing about AppData with you by email? (yes/no)

(this comment will be deleted)

**Comment by hughsie** [comment]
Sure, no problem. Use richard_at_hughsie.com to discuss.

Richard
